src/main/java/org/ow2/proactive/procci/model/occi/metamodel/ResourceBuilder.java

```diff
@@ -0,0 +1,126 @@
+import org.ow2.proactive.procci.model.occi.metamodel.constants.Kinds;
+import org.ow2.proactive.procci.model.occi.metamodel.rendering.ResourceRendering;
+import org.ow2.proactive.procci.model.utils.ConvertUtils;
+import org.ow2.proactive.procci.service.occi.MixinService;
+import lombok.Getter;
+
+import static org.ow2.proactive.procci.model.occi.metamodel.constants.Attributes.ENTITY_TITLE_NAME;
+import static org.ow2.proactive.procci.model.occi.metamodel.constants.Attributes.ID_NAME;
+import static org.ow2.proactive.procci.model.occi.metamodel.constants.Attributes.SUMMARY_NAME;
+
+@Getter
+public class ResourceBuilder {
+
+
+    protected Optional<String> url;
+    protected Optional<String> title;
+    protected List<Mixin> mixins;
+    protected Optional<String> summary;
+    protected List<Link> links;
+
+    public ResourceBuilder() {
+        this.url = Optional.empty();
+        this.title = Optional.empty();
+        this.mixins = new ArrayList<>();
+        this.summary = Optional.empty();
+        this.links = new ArrayList<>();
+    }
+
+    public ResourceBuilder(Model cloudAutomation) {
+
+        Map<String, String> attributes = cloudAutomation.getVariables();
+
+        this.url = Optional.ofNullable(attributes.get(ID_NAME));
+        this.title = Optional.ofNullable(attributes.get(ENTITY_TITLE_NAME));
+        this.summary = Optional.ofNullable(attributes.get(SUMMARY_NAME));
+
+        this.mixins = new ArrayList<>();
+        this.links = new ArrayList<>();
+    }
+
+    public ResourceBuilder(MixinService mixinService,
+            ResourceRendering rendering) throws ClientException {
+        this.url = Optional.ofNullable(rendering.getId());
+        this.title = ConvertUtils.convertStringFromObject(Optional.ofNullable(rendering.getAttributes())
+                .map(attributes -> attributes.getOrDefault(ENTITY_TITLE_NAME, null)));
+        this.summary = ConvertUtils.convertStringFromObject(Optional.ofNullable(
+                rendering.getAttributes()).map(attributes -> attributes.getOrDefault(SUMMARY_NAME, null)));
+        this.mixins = new ArrayList<>();
+        for (String mixin : Optional.ofNullable(rendering.getMixins()).orElse(new ArrayList<>())) {
+            this.mixins.add(mixinService.getMixinByTitle(mixin));
+        }
+        associateProviderMixin(mixinService, rendering.getAttributes());
+        this.links = new ArrayList<>();
+    }
+
+    /**
+     * Check all attributes and add in the mixin collection the attributes from mixin
+     *
+     * @param attributes is the attriutes list of the compute
+     * @throws ClientException is thrown if there is an error during the mixin reading
+     */
+    void associateProviderMixin(MixinService mixinService,
+            Map<String, Object> attributes) throws ClientException {
+        if (attributes == null) {
+            return;
+        }
+        for (String mixinName : attributes.keySet()) {
```

Comment:
Why not using java8 syntax (if attributes is not so huge, do not use it due to the overhead)
items.stream()
	.filter( ... )
	.forEach( ... )